search/includes/classes/class-versioning.php

```diff
@@ -200,7 +201,19 @@ public function get_inactive_versions( Indexable $indexable ) {
 	 * @return array Array of index versions
 	 */
 	public function get_versions( Indexable $indexable, bool $provide_default = true ) {
-		$versions = get_option( self::INDEX_VERSIONS_OPTION, array() );
+		$sources = [
+			'per_site_versions' => get_option( self::INDEX_VERSIONS_OPTION, array() ),
+			'deprecated_multisite_storage' => get_site_option( self::INDEX_VERSIONS_OPTION, array() ),
+			'global_versions' => get_site_option( self::INDEX_VERSIONS_OPTION_GLOBAL, array() ),
+		];
+
+		$versions = [];
+
+		foreach ( $sources as $source ) {
+			if ( is_array( $source ) ) {
+				$versions = array_merge( $versions, $source );
+			}
+		}
```

Comment:
This I think should handle all the use cases, when mix of deprecated and new versions are on site.  Due to the order the globals have overwrite the local settings (which is important when working with old way of storing user - per site locally vs now one per network globally.).

Please let me know if you can see any issues with this approach!

Reply:
@pschoffer there may be a better way to do this. Firstly, we should be checking `$indexable->global` to see where we should pull the data from. Then, if the indexable _is_ global, we can have the fallback to the deprecated option if the right site option doesn't exist or is empty. When this occurs, we should log it so we can see if it's actually affecting any sites (and if not, remove the fallback).

Reply:
Good idea with getting it based on the global flag!

The fallback needs to stay for both globals and non-global I think. Because the code before would store the versions on network option on multisite no matter if it is a global or non-global indexable.

I also added the logging and a call to update_versions so that it gets stored in the correct place for the next time. This should ensure that the deprecated location can be phased out eventually.